catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.view.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+ /*global require*/
+ var Marionette = require('marionette');
+ var _ = require('underscore');
+ var $ = require('jquery');
+ var template = require('./search-form.hbs');
+ var CustomElements = require('js/CustomElements');
+ var user = require('component/singletons/user-instance');
+
+ module.exports = Marionette.LayoutView.extend({
+    template: template,
+    tagName: CustomElements.register('search-form'),
+    className: 'is-button',
+    events: {
+        'click': 'changeView'
+    },
+    initialize: function(options) {
+        this.queryModel = options.queryModel;
+    },
+    onRender: function() {
+        if (this.model.get('type') === 'basic' || this.model.get('type') === 'text') {
+            this.$el.addClass('is-static');
+        }
+    },
+    changeView: function() {
+        this.queryModel.set('formTemplate', this.model.get('formTemplate'));
```

Comment:
We should be saving the id of the template instead, formTemplate isn't guaranteed to be static over time.